cmd/helm/root.go

```diff
@@ -155,7 +158,13 @@ func newRootCmd(actionConfig *action.Configuration, out io.Writer, args []string
 
 	registryClient, err := newDefaultRegistryClient(false)
 	if err != nil {
-		return nil, err
+		// see issues #11009
+		debug("generate registry client failed: %v", err.Error())
+		for i := 0; len(args) > 0 && i < len(CMDNeedRegistryClient); i++ {
```

Comment:
can we separate this "ignore" functionality into a helper function please. Ideally we might write something like:

```
if err != nil && !ignoreCommandNewRegistryClientError(cmd.Name()) { // Note: I'm not exactly sure how to get the invoked (sub)command from Cobra, as my below comment
    return nil, err
}
```

This `ignoreCommandNewRegistryClientError` function should likely have test case(s).